pallets/exchange/src/lib.rs

```diff
@@ -92,8 +92,8 @@ pub mod pallet {
 				Self::process_exchange_intentions(&pair_account, &mut asset_a_ins, &mut asset_b_ins);
 			}
 
-			ExchangeAssetsIntentionCount::<T>::remove_all(None);
-			ExchangeAssetsIntentions::<T>::remove_all(None);
+			let _ = <ExchangeAssetsIntentionCount<T>>::clear(u32::MAX, None);
```

Comment:
why u32::MAX?

Reply:
There needs to be some `u32` limit.

Reply:
you have let _ here too , doesnt clippy complain ?  like in the comment below  about the creat pool handler ? 

and what is the limit for ? ( i could look it up, i know;p))

Reply:
It's because the return type of `clear` has [this](https://github.com/paritytech/substrate/blob/720093428b90363cb909162ef7c05e5bd51ae626/primitives/externalities/src/lib.rs#L57) attribute. In the case of `T::AMMHandler::on_create_pool(asset_pair.asset_in, asset_pair.asset_out)?;`, there are two possibilities. The result is "used" by `?` or the attribute is not implemented for unit type that we get after applying `?`. Not sure about this.

`limit` limits the number of deletions.